doc/changelog.md

```diff
@@ -0,0 +1,919 @@
+# Changelog
+
+Listed here are the changes between each release of SmartSim,
+SmartRedis and SmartDashboard.
+
```

Comment:
Are there supposed to be links to the SmartRedis and SmartDashboard changelogs here?